pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -306,14 +308,32 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 			return nil, httperror.WrapAPIError(err, httperror.ServerError, "Error processing clusterTemplate answers")
 		}
 		keyParts := strings.Split(question.Variable, ".")
-		values.PutValue(dataFromTemplate, val, keyParts...)
+		if strings.HasPrefix(question.Variable, registryConfigPath) {
+			key := keyParts[len(keyParts)-1]
+			index, err := strconv.Atoi(string(question.Variable[len(registryConfigPath)+1]))
```

Comment:
Does this work for double-digit indexes? Do we need to support double-digit indexes?

Reply:
Technically it could be a two-digit index, but in practice, the index should be never greater than 1 as Rancher UI only allows to set one private registry for your cluster. 
In my option, treating the index as single-dight should be sufficient. What do you think? @thedadams

Reply:
Update: the PR is updated to handle the index regardless of its many digits.